modules/light-clients/07-tendermint/types/update.go

```diff
@@ -261,3 +261,48 @@ func update(ctx sdk.Context, clientStore sdk.KVStore, clientState *ClientState,
 
 	return clientState, consensusState
 }
+
+// CheckForMisbehaviour detects duplicate height misbehaviour and BFT time violation misbehaviour
+func (cs ClientState) CheckForMisbehaviour(ctx sdk.Context, cdc codec.BinaryCodec, clientStore sdk.KVStore, msg exported.ClientMessage) bool {
+	switch msg := msg.(type) {
+	case *Header:
+		tmHeader := msg
+		consState := tmHeader.ConsensusState()
+
+		// Check if the Client store already has a consensus state for the header's height
+		// If the consensus state exists, and it matches the header then we return early
+		// since header has already been submitted in a previous UpdateClient.
+		prevConsState, _ := GetConsensusState(clientStore, cdc, tmHeader.GetHeight())
+		if prevConsState != nil {
+			// This header has already been submitted and the necessary state is already stored
+			// in client store, thus we can return early without further validation.
+			if reflect.DeepEqual(prevConsState, tmHeader.ConsensusState()) {
+				return false
+			}
+
+			// A consensus state already exists for this height, but it does not match the provided header.
+			// The assumption is that Header has already been validated. Thus we can return true as misbehaviour is present
+			return true
```

Comment:
My linter here suggests this should be simplified to: 
```
return !reflect.DeepEqual(prevConsState, tmHeader.ConsensusState())
```

Any objections to changing this and I can adjust the comments as necessary?

Reply:
I actually find this suggestion harder to parse than what exists currently.

Reply:
Same tbh, it always is when it's the negation of some boolean expression. Happy to leave as is, I wonder if we can suppress the static-check linter here.

Reply:
I'd prefer to leave as is